ODIN_II/SRC/netlist_statistic.cpp

```diff
@@ -69,6 +69,24 @@ void init_stat(netlist_t* netlist) {
     netlist->num_logic_element = 0;
 }
 
+void mixing_optimization_stats(nnode_t* node, netlist_t* netlist) {
+    uintptr_t traverse_mark_number = 1010;
```

Comment:
Use the pointer shenanigans , hardcoded number might collide, or better would be to have a separate PR with a unique traversal id generator and have it generated in a more elegant fashion than this

Reply:
Hello @jeanlego,  I do not understand what pointer shenanigans are, could you please explain it a little bit better? I can imagine a collision scenario you are talking about, but the code you have provided as an example in your previous comment does not type-check (it errors during the build process, unless I made a mistake copying it). Also, could you please elaborate on the PR for traversal id generators? Is it a requirement for this PR to get in?

Reply:
As stated in one of the previous comment:

```
static const int traversal_id = 0;
static const uintptr_t mult_optimization_traverse_value = &traversal_id;
```

The PR is not mandatory to uniformize generating a unique traversal id, more of a "if you feel generous" :p

Reply:
> 
> 
> As stated in one of the previous comment:
> 
> ```
> static const int traversal_id = 0;
> static const uintptr_t mult_optimization_traverse_value = &traversal_id;
> ```
> 
> The PR is not mandatory to uniformize generating a unique traversal id, more of a "if you feel generous" :p

A typecast was missing there! Unless it's intentional. Regarding courtesy PR's, I have a stack of PR's planned, so I will consider adding the generator to there. But that's subject of a later discussion. Thank you!